### ISSUE TITLE
Corrupted display name when passing c_str() to begin()

### ISSUE THREAD

**Issue by kafana** [HomeSpan/HomeSpan#837]
I am attempting to dynamically set up the display name using the `homespan.begin(...)` function. However, when I pass the `c_str()` of a String object into the `begin` function, it seems to affect the character pointer, resulting in the display name being corrupted. I assume that the `begin()` function should not modify the display name. This functionality was working correctly in version 1.8.

This is example code:

```
char nameSuffix[20];
sprintf(nameSuffix, "%06d", 101);
String displayName = "My Device Number " + String(nameSuffix);

...

homeSpan.begin(Category::Lighting, displayName.c_str());
```

The output looks something like this:

```
Accessory configuration number: 1

l��? is READY!

Trying to connect to XXXXXX_XXXXXX.  Waiting 1 sec...
Trying to connect to XXXXXX_XXXXXX.  Waiting 2 sec...
E (7391) wifi:sta is connecting, return error
WiFi Connected!  IP Address = xxx.21.31.205

Starting MDNS...

HostName:      HomeSpan-CAE729DD642F.local:80
Display Name:  l��?
Model Name:    HomeSpan-ESP32
Setup ID:      HSPN
```

Passing a constant string into `begin()` call works perfectly fine.

```
homeSpan.begin(Category::Lighting, "My Device 101");
```

```
My Device 101 is READY!

Trying to connect to XXXXXX_XXXXXX.  Waiting 1 sec...
WiFi Connected!  IP Address = xxx.21.31.205

Starting MDNS...

HostName:      HomeSpan-CAE729DD642F.local:80
Display Name:  My Device 101
Model Name:    HomeSpan-ESP32
Setup ID:      HSPN
```

Thank you for the amazing library!

**Comment by HomeSpan** [comment]
This is because the `homeSpan.begin()` function does not copy the full character array for any names passed to it (though I probably should do that in a future update). It only copies the pointers.

This means that if String is declared within `setup()`, it will go out of scope once the `setup()` function is completed by the time `homeSpan.poll()` is called within the main `loop()`, the pointer to displayName will likely no longer be valid.  Depending on the sketch, the memory occupied by displayName may or may not still contain a valid character array.  If this worked in 1.8.0, it was only by luck.

The workaround is to define your String as a global variable instead of within setup so the name is retained and can be used once homeSpan.poll() is called.  I'll see about reserving space to copy the full string in a future update.

**Comment by kafana** [comment]
> This means that if String is declared within `setup()`, it will go out of scope once the `setup()` function is completed by the time `homeSpan.poll()` is called within the main `loop()`, and the pointer to displayName will likely no longer be valid. Depending on the sketch, the memory occupied by displayName may or may not still contain a valid character array. If this worked in 1.8.0, it was only by luck.

Thank you for the clarification. I've been using the same pattern for both serial numbers and pairing codes. I will transfer all related data to a global scope until a permanent fix is implemented.

**Comment by HomeSpan** [comment]
I've updated the logic in `homeSpan.begin()` and `homeSpan.setHostNameSuffix()` to copy the specified strings instead of copying the pointer.  This should solve the problem above and allow you to use strings that go out of scope without any issue.  New code is in the dev branch and will be rolled out in next release.

**Comment by kafana** [comment]
Thank you so much for addressing this promptly. When do you plan to release v1.10.0?

**Comment by HomeSpan** [comment]
Will release in probably a few more weeks.  Working out some final tweaks.

**Comment by HomeSpan** [comment]
Note the new version will be 1.9.1 (there is a branch for it, though it's not yet been tagged for release since I'm still making changes).

**Comment by HomeSpan** [comment]
These changes have now been released into production as version 1.9.1.
